pkg/kubelet/status/status_manager.go

```diff
@@ -402,9 +404,19 @@ func (m *manager) updateStatusInternal(pod *v1.Pod, status v1.PodStatus, forceUp
 	}
 
 	normalizeStatus(pod, &status)
+
 	// The intent here is to prevent concurrent updates to a pod's status from
 	// clobbering each other so the phase of a pod progresses monotonically.
-	if isCached && isPodStatusByKubeletEqual(&cachedStatus.status, &status) && !forceUpdate {
+	shouldUpdate := forceUpdate
+	if !shouldUpdate && m.lastUpdateTime != nil {
+		// if the last update is too old, we need to update status
+		// to handle the situation of connectivity being returned to the node.
+		if time.Now().Sub(*m.lastUpdateTime) > 200*time.Second {
+			shouldUpdate = true
+		}
+	}
+
```

Comment:
Need a unit test to verify the change, and to have a reasonable unit test, you may want to substitute the clock used here by the one in `k8s.io/apimachinery/pkg/util/clock`